### ISSUE TITLE
JDBC password provided via environment variable FLYWAY_PASSWORD does no longer seem to work

### ISSUE THREAD

**Issue by jerdaane** [flyway/flyway#3837]
###### Which version and edition of Flyway are you using?
10.7.1

###### Which client are you using? (Command-line, Java API, Maven plugin, Gradle plugin)
Command-line / flyway docker image

###### Which database are you using? (Type & version)
Postgres 14.10

###### Which operating system are you using?
Alpine (docker image)

###### What did you do? (Please include the content causing the issue, any relevant configuration settings, the SQL statement(s) that failed (if any), and the command you ran)
Flyway is executed via the official docker image lyway/flyway:10.7.0-alpine. The database password is provided via the FLYWAY_PASSWORD environement variable.
For instance this is in my docker compose file:
```
  flyway:
    build: ./../flyway
    command: -url=jdbc:postgresql://postgres:5432/xyz -user=postgres -connectRetries=60 migrate
    environment:
      TZ: Europe/Amsterdam
      FLYWAY_PASSWORD: postgres
```
The docker file is based on `FROM flyway/flyway:10.7.1-alpine`

The above works fine for a previous version (10.6.0) but for 10.7.1 the following errors from flyway appear:
```
WARNING: Connection error: The server requested SCRAM-based authentication, but no password was provided.
```

###### What did you expect to see?
The flyway migrations to be executed. According to[ the documentation](https://github.com/flyway/flyway/blob/43b4a835cd0472a4c7ecd5116280d6a3a9ad3804/documentation/Flyway%20CLI%20and%20API/Configuration/Parameters/Environments/Password.md?plain=1#L39) this should be possible.

###### What did you see instead?
Error messages that no connection with the database can be made

**Comment by dennisgsmith** [comment]
I'm experiencing the same issue after updating to 10.7.1. However I'm not using alpine, just the vanilla Docker image.
```yaml
  migrate:
    build:
      context: migrate
    restart: on-failure
    environment:
      - FLYWAY_EDITION=community
      - FLYWAY_LOCATIONS=filesystem:/flyway/migrate
      - FLYWAY_PASSWORD=postgres
      - FLYWAY_SCHEMAS=public
      - FLYWAY_URL=jdbc:postgresql://db:5432/postgres
      - FLYWAY_USER=postgres
    depends_on:
      db:
        condition: service_healthy
```
```Dockerfile
FROM flyway/flyway:10.7
COPY . /flyway/migrate/
ENTRYPOINT [ "flyway" ]
CMD [ "-connectRetries=60", "migrate" ]
```
logs:
```
migrate-1  | See release notes here: https://rd.gt/416ObMi
migrate-1  | ERROR: Unable to connect to the database. Configure the url, user and password!
migrate-1  | Flyway OSS Edition 10.7.1 by Redgate
```
Reverting to 10.6 fixes.

**Comment by Peter-Laws** [comment]
Hello, this should now be addressed in today's release v10.7.2, please could you update and verify?
([release notes](https://documentation.red-gate.com/flyway/release-notes-and-older-versions/release-notes-for-flyway-engine))

**Comment by dennisgsmith** [comment]
For the sake of the original issue I also tested with the alpine image and didn't run into any problems.
